recipes/libdeflate/all/test_package/conanfile.py

```diff
@@ -6,6 +6,15 @@ class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
     generators = "cmake", "cmake_find_package_multi"
 
+    def build_requirements(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # Workaround for CMake bug with error message:
+            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
+            # set. This could be because you are using a Mac OS X version less than 10.5
+            # or because CMake's platform configuration is corrupt.
+            # FIXME: Remove once CMake on macOS/M1 CI runners is upgraded.
```

Comment:
MacOS runners are using CMake 3.15 following the requirement here: https://github.com/conan-io/tribe/blob/main/design/004-tools-cmake.md. If it is not enough, it's not going to be upgraded because of that restriction.

We can have a look to the other reasons you mention.

Reply:
I took the trick from another recipe actually. I guess it's a bug in CMake with relocatable shared libs on macos M1, fixed recently.

Reply:
I added this temporary workaround, it's more or less explained in #7146
it's a CMake bug that was fixed only in 3.20.1+

Reply:
Indeed: https://github.com/conan-io/conan-center-index/pull/7146#issuecomment-912293457

Reply:
It's amazing the number of details you manage to keep in your head 😊 .